mrtarget/modules/ChEMBL.py

```diff
@@ -201,19 +194,13 @@ def _append_to_mol2syn(m2s_dict, molecule):
             self._logger.warn("No molecules in set")
             return
 
-        #build a URL to query chembl for
-        url = self.molecule_set_uri_pattern.format(';'.join(molecule_set))
-
-        #actually query that URL and get json back
-        data = None
-        try:
-            response = requests.get(url)
-            response.raise_for_status()
-            data = response.json()
-        except Exception as e:
-            self._logger.error('problem downloading molecule info from url %s', url)
-            #re-raise exception to allow propegation
-            raise e
+        data = {'molecules':[]}
+        for mol_k in molecule_set:
+            if self.molecules_dict.has_key(mol_k):
+                data['molecules'].append(self.molecules_dict[mol_k])
+            else:
+                self._logger.error('problem retrieving the molecule info from the local db %s', str(mol_k))
+                raise ValueError('problem retrieving the molecule info from the local db', str(mol_k))
 
         #if the data is what we expected, process it
         if 'molecules' in data:
```

Comment:
Given the key is being created just before, why is this code checking for it?

Reply:
The item must be present otherwise something is missing from ChEMBL molecule REST API.